contracts/simple_dex/lib.rs

```diff
@@ -365,6 +357,50 @@ mod simple_dex {
                 .map_err(|why| DexError::InkEnv(format!("Can't retrieve own code hash: {:?}", why)))
         }
 
+        /// Swap trade output given a curve with equal token weights
+        ///
+        /// swap_fee_percentage (integer) is a percentage of the trade that goes towards the pool
+        /// B_0 - (100 * B_0 * B_i) / (100 * (B_i + A_i) -A_i * fee)
+        #[ink(message)]
+        pub fn out_given_in(
```

Comment:
this could call into a stateless _out_given_in (or  similar name) that we could unit test, if we narrow down the precision problems to it (likely to happen)? Else LGTM

Reply:
:+1: But I propose a new PR for that.

Reply:
I hold you to that